climada_petals/hazard/tc_surge_bathtub.py

```diff
@@ -119,17 +122,16 @@ def from_tc_winds(wind_haz, topo_path, inland_decay_rate=0.2, add_sea_level_rise
         coastal_centroids_h -= add_sea_level_rise
 
         # Efficient way to subtract from selected columns of sparse csr matrix
-        nz_coastal_cents = inten_surge[:,coastal_idx].nonzero()
-        nz_coastal_cents_inten = (nz_coastal_cents[0], coastal_idx[nz_coastal_cents[1]])
-        inten_surge[nz_coastal_cents_inten] -= coastal_centroids_h[nz_coastal_cents[1]]
+        inten_surge.data -= coastal_centroids_h[cent_to_coastal_idx[inten_surge.indices]]
```

Comment:
This is a comparably minor improvement, but it's also preventing that any temporary arrays are created that are larger than the amount of non-zero data stored in the sparse matrix.